test/common/stats/stats_impl_test.cc

```diff
@@ -466,5 +469,23 @@ TEST(TagProducerTest, CheckConstructor) {
       "No regex specified for tag specifier and no default regex for name: 'test_extractor'");
 }
 
+TEST(SomeTest, HotAllocConsistencyTest) {
```

Comment:
@mrice32 I think I don't have a great way to test this twice, once in the context of injecting shared memory, and once in the context of injecting non-shared memory. Is there a test we can write to prove this works in the original worry case in https://github.com/envoyproxy/envoy/issues/2453 ?